ext/ffi/lib.rs

```diff
@@ -756,44 +558,421 @@ fn op_ffi_get_static(
+    call_args
+      .push(value_as_arg(scope, arg, symbol.parameter_types[idx]).unwrap());
+  }
+
+  let value = match symbol.result_type {
+    NativeType::Void => {
+      unsafe { symbol.cif.call::<()>(symbol.ptr, &call_args) };
+      return;
+    }
+    NativeType::U8 => {
+      let value = unsafe { symbol.cif.call::<u8>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::I8 => {
+      let value = unsafe { symbol.cif.call::<i8>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::U16 => {
+      let value = unsafe { symbol.cif.call::<u16>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::I16 => {
+      let value = unsafe { symbol.cif.call::<i16>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::U32 => {
+      let value = unsafe { symbol.cif.call::<u32>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::I32 => {
+      let value = unsafe { symbol.cif.call::<i32>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::U64 => {
+      let value = unsafe { symbol.cif.call::<u64>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::I64 => {
+      let value = unsafe { symbol.cif.call::<i64>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::USize => {
+      let value = unsafe { symbol.cif.call::<usize>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::ISize => {
+      let value = unsafe { symbol.cif.call::<isize>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::F32 => {
+      let value = unsafe { symbol.cif.call::<f32>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::F64 => {
+      let value = unsafe { symbol.cif.call::<f64>(symbol.ptr, &call_args) };
+      serde_v8::to_v8(scope, value).unwrap()
+    }
+    NativeType::Pointer => {
+      let value =
+        unsafe { symbol.cif.call::<*const u8>(symbol.ptr, &call_args) } as u64;
+      let ptr = U32x2::from(value);
+      serde_v8::to_v8(scope, ptr).unwrap()
+    }
+  };
+  rv.set(value);
+}
+
+pub fn op_ffi_call_fn(
+  scope: &mut deno_core::v8::HandleScope,
+  args: deno_core::v8::FunctionCallbackArguments,
+  mut rv: deno_core::v8::ReturnValue,
+) {
+  // SAFETY: deno_core guarantees args.data() is a v8 External pointing to an OpCtx for the isolates lifetime
+  let ctx = unsafe {
+    &*(v8::Local::<v8::External>::cast(args.data().unwrap_unchecked()).value()
+      as *const deno_core::_ops::OpCtx)
+  };
+  let rid = args.get(0).uint32_value(scope).unwrap();
+  let symbol = args.get(1).to_string(scope).unwrap();
+  let symbol = symbol.to_rust_string_lossy(scope);
```

Comment:
Hmmm, probably a totally irrelevant performance optimisation but I wonder if the symbols could be kept in a Map keyed by the V8's internal UTF-16 representation to avoid the need to convert between V8 strings and Rust strings...